tornjak-frontend/src/components/spiffe-helper.js

```diff
@@ -83,23 +81,24 @@ class SpiffeHelper extends Component {
         return metadata[0]
       }
     }
-    return {"plugin":"", "cluster":""}
+    return { "plugin": "", "cluster": "" }
   }
 
   // getAgentEntries provides an agent and a list of entries and returns
   // the list of entries which are associated with this agent
-  getAgentEntries (agent, entries) {
-      let nodeEntries = entries.filter(e => e.parent_id.path === "/spire/server")
-      let agentSelectors = new Set(agent.selectors.map(formatSelectors))
-      let isAssocWithAgent = e => {
-          let entrySelectors = new Set(e.selectors.map(formatSelectors))
-          return isSuperset(agentSelectors, entrySelectors)
-      }
-
-      return nodeEntries.filter(isAssocWithAgent)
+  getAgentEntries(agent, entries) {
+    let nodeEntries = entries.filter(e => e.parent_id.path === "/spire/server")
+    if(agent === undefined || agent[0].selectors === undefined) 
+      return [];
+    let agentSelectors = new Set(agent[0].selectors.map(formatSelectors))
```

Comment:
why is this an array or things? It should only accept a singular agent.....

Reply:
it only accepts a singular agent, it's just filtered agent so it's an agent in an array. Cleaned it up.